torchvision/io/_video_opt.py

```diff
@@ -438,15 +447,20 @@ def _probe_video_from_memory(video_data):
     return info
 
 
-def _convert_to_sec(start_pts, end_pts, pts_unit, time_base):
+def _convert_to_sec(start_pts: float, end_pts: float, pts_unit: str, time_base: Fraction) -> Tuple[float, float, str]:
     if pts_unit == 'pts':
         start_pts = float(start_pts * time_base)
         end_pts = float(end_pts * time_base)
         pts_unit = 'sec'
     return start_pts, end_pts, pts_unit
 
 
-def _read_video(filename, start_pts=0, end_pts=None, pts_unit="pts"):
+def _read_video(
+    filename: str,
+    start_pts: int = 0,
+    end_pts: Optional[float] = None,
```

Comment:
`end_pts` can be int, float or Fraction. So, this should be Optional[Union[float, Fraction]].

Reply:
Apart from if I'm mistaking, `Union` are not yet allowed to ensure torchscript is still working :/

Reply:
pytorch/pytorch#64234 added support for `Union`'s so feel free to use them now!

Reply:
Holy cow, and I wasn't notified ?? :joy:
Let's clean all those annotations then :smile: I'll deal with it shortly!

Reply:
My apologies, I forgot about the "shortly" along the way, I just dealt with it :sweat_smile: